app/controllers/api_controller/parser.rb

```diff
@@ -133,8 +133,35 @@ def parse_id(resource, collection)
       resource["id"].kind_of?(Integer) ? resource["id"] : nil
     end
 
+    def parse_by_attr(resource, type, attr_list)
+      klass = collection_class(type)
+      attr_list.each do |attr|
```

Comment:
not really as I need the resultant obj.id, but maybe a select/map combo.  let me try something.